arch/xtensa/Kconfig

```diff
@@ -153,6 +153,33 @@ config XTENSA_HIFI_SHARING
 	  across context switches to allow multiple threads to perform
 	  concurrent HiFi operations.
 
+if XTENSA_HIFI_SHARING
+
+choice XTENSA_HIFI_SHARING_MODEL
+	prompt "Xtensa HiFi Sharing Model"
+	depends on XTENSA_HIFI_SHARING
+	default XTENSA_EAGER_HIFI_SHARING
+
+config XTENSA_EAGER_HIFI_SHARING
+	bool "Eager HiFi Sharing"
+	help
+	  This option enables eager sharing of HiFi registers across context
+	  switches. This means that the HiFi registers are unconditionally
+	  saved and restored on every context switch, allowing multiple threads
+	  to use HiFi instructions concurrently.
+
+config XTENSA_LAZY_HIFI_SHARING
+	bool "Lazy HiFi Sharing"
+	depends on SCHED_CPU_MASK_PIN_ONLY || (MP_MAX_NUM_CPUS = 1)
+	help
+	  This option enables lazy sharing of HiFi registers across context
+	  switches. This means that the HiFi registers are saved and restored
+	  only when a thread actually uses HiFi instructions, allowing for
+	  more efficient use of resources when HiFi instructions are not used.
```

Comment:
I assume this would work when we have a userspace application thread (that uses Hifi) being preempted by kernel (that does not use Hifi) ?